### PR TITLE
[Data Processing UX] Allow user to change settings of multiple filters without running QC pipeline

### PR DESCRIPTION
**Pull request by StefanBabukov**
# Background
#### Link to issue 
https://biomage.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=BIOMAGE&modal=detail&selectedIssue=BIOMAGE-1122
#### Link to staging deployment URL 
https://ui-grasp6-ui345.scp-staging.biomage.net/
#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR
image

### CODE REVIEW

**Comment by StefanBabukov** [comment]
For the Data integration and Configure embedding steps, should I still keep them separated from the other filters when saving changes - there is no "Save changes" button but instead a "Run" for each of them. Example:
 When the user presses `Run` on Data integration and has changed other things in Cellsizedistribution, should the changes in cellsizedistribution be also applied @vickymorrison @aerlaut ?

**Comment by vickymorrison** [comment]
> For the Data integration and Configure embedding steps, should I still keep them separated from the other filters when saving changes - there is no "Save changes" button but instead a "Run" for each of them. Example:
> When the user presses `Run` on Data integration and has changed other things in Cellsizedistribution, should the changes in cellsizedistribution be also applied @vickymorrison @aerlaut ?

Yes. Clicking 'run' in Data Integration would apply all the saved chenges across all data processing steps.

**Comment by StefanBabukov** [comment]
Also I can see if I go to Configure embedding and change something - clicking Run afterwards doesn't trigger the pipeline again is this on purpose?

**Comment by vickymorrison** [comment]
> Also I can see if I go to Configure embedding and change something - clicking Run afterwards doesn't trigger the pipeline again is this on purpose?

@ogibson-biomage Can you answer this question from Stefan?
My understanding is that changing something in Configure Embedding re-computes the embedding but the whole pipeline does not have to re-run.

**Comment by aerlaut** [comment]
> > Also I can see if I go to Configure embedding and change something - clicking Run afterwards doesn't trigger the pipeline again is this on purpose?
> 
> @ogibson-biomage Can you answer this question from Stefan?
> My understanding is that changing something in Configure Embedding re-computes the embedding but the whole pipeline does not have to re-run.

I had a discussion on this with Stefan yesterday: 
- If there are changes to the filter settings before configure embedding, clicking Run will run the whole pipeline again. This has to be done because the cell sets may change due to the changes in filter settings.
- If there are no changes on the filter settings before configure embedding, clicking Run will only run the configure embedding, without having to run the QC pipeline again (as it is now).

This may lead to ambiguous UX on the behaviour of the Run button. I've added pointer in ticket 1125 https://biomage.atlassian.net/browse/BIOMAGE-1125 so that clicking the Run button in Configure Embedding with changes to the settings of the previous filter will also trigger the reminder popup modal.

File: src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx [hunk 1]
```diff
@@ -119,6 +118,11 @@ const CalculationConfig = (props) => {
     updateSettings(changes);
     setChangesOutstanding(false);
     dispatch(loadEmbedding(experimentId, embeddingMethod, true));
+    if (changedFilters?.current.size) {
+      // other steps are changed so we run the pipeline
+      changedFilters.current.add(FILTER_UUID);
+      onPipelineRun();
+    }
```
Comment (aerlaut) [inline src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx]: Maybe this should be an `if/else`? If there are changes to the previous filters, `dispatch(loadEmbedding())` doesn't need to be called before running the pipeline.

**Review by aerlaut — COMMENTED** [review]
Other than my comment, everything else looks good. Can you add a test to check if `changedFilters` is handled correctly in `configureEmbedding`? i.e. pipeline is run if there are changed filters, else only `dispatch` to `loadEmbedding()` is run.

**Review by vickymorrison — APPROVED** [review]

**Pull request merged**